google/cloud/aiplatform/training_jobs.py

```diff
@@ -103,6 +103,65 @@ def __init__(
         self._project = project
         self._gca_resource = None
 
+    @property
+    @abc.abstractmethod
+    def _supported_training_schemas(self) -> List[str]:
+        """List of supported schemas for this training job"""
+
+        pass
+
+    @classmethod
+    def get(
+        cls,
+        resource_name: str,
+        project: Optional[str] = None,
+        location: Optional[str] = None,
+        credentials: Optional[auth_credentials.Credentials] = None,
+    ) -> "CustomTrainingJob":
+        """Get CustomTrainingJob for the given resource_name.
+
+        Args:
+            resource_name (str):
+                Required. A fully-qualified resource name or ID.
+            project (str):
+                Optional project to retrieve dataset from. If not set, project
+                set in aiplatform.init will be used.
+            location (str):
+                Optional location to retrieve dataset from. If not set, location
+                set in aiplatform.init will be used.
+            credentials (auth_credentials.Credentials):
+                Custom credentials to use to upload this model. Overrides
+                credentials set in aiplatform.init.
+
+        Raises:
+            ValueError: If the retrieved training job's training task definition
+                doesn't match the custom training task definition.
+
+        Returns:
+            An AI Platform Training Job
+        """
+
+        # Create job with dummy parameters
```

Comment:
I created a bug to extract project and location from resource name (b/180923329) if given using this parser: https://github.com/googleapis/python-aiplatform/blob/b7aeeb637b7c36792a2de0b65c868121a4a7e412/google/cloud/aiplatform_v1beta1/services/pipeline_service/client.py#L197

We can add as a follow up for all classes.